favourite-colour-java/src/main/java/com/github/simplesteph/udemy/kafka/streams/FavouriteColourApp.java

```diff
@@ -50,18 +50,18 @@ public static void main(String[] args) {
         KTable<String, Long> favouriteColours = usersAndColoursTable
                 // 5 - we group by colour within the KTable
                 .groupBy((user, colour) -> new KeyValue<>(colour, colour))
-                .count("CountsByColours");
+                .count(Materialized.as("CountsByColours"));
```

Comment:
This one does not work, I ran the example and it fails at runtime with a serialisation error. Remember the default serialiser is String and now you get a Long.

This one was a bit painful to figure out, but use this:

```
Serde<String> stringSerde = Serdes.String();
        Serde<Long> longSerde = Serdes.Long();

        // step 3 - we count the occurences of colours
        KTable<String, Long> favouriteColours = usersAndColoursTable
                // 5 - we group by colour within the KTable
                .groupBy((user, colour) -> new KeyValue<>(colour, colour))
                .count(Materialized.<String, Long, KeyValueStore<Bytes, byte[]>>as("CountsByColours")
                        .withKeySerde(stringSerde)
                        .withValueSerde(longSerde));
```

Reply:
I will update accordingly, I dont have access to course's sample test-data though.

Reply:
No need for sample data, just push one sample message and if there's no runtime error it's good ! You can easily figure out what data might look like :)